aerogear/mobile-security-service-operator
# feat(Bind):  Rename the BIND type for APP (AEROGEAR-9090)

## Motivation
https://issues.jboss.org/browse/AEROGEAR-9090

## What
- Rename the type Bind to App 

PS.: the unbind will be removed and the same logic will be a finalizer of the app type. 

## How

1. Using the operator-sdk tool to create the new type and files
2. Moving the logic for the new type
3. Removing all BIND files/dirs
4. Updating Make file and ReadMe

## Verification Steps
1. Clone this PR and use the image "cmacedo/mobile-security-service-operator:AEROGEAR-9090-dev" by updating the `operator.yaml` file
2. Use the command `make create-all` when all be up and running
3. Create a new namespace `oc new-project appptest`
4. Update the APP CR with the master VIP (`kubectl cluster-info`)
5. Use the command `make create-app` (check if the app was created in the Rest Service and if the SDKconfigMap was created in the app namespace)
6. Check if the related points are properly updated into the README (note that just the name changed)

## Checklist:


## Progress


## Additional Notes

<img width="1599" alt="Screenshot 2019-04-26 at 17 56 05" src="https://user-images.githubusercontent.com/7708031/56823758-a960b000-684c-11e9-91ad-696d0d587a95.png">

<img width="1075" alt="Screenshot 2019-04-26 at 17 57 26" src="https://user-images.githubusercontent.com/7708031/56823822-db721200-684c-11e9-942f-9b33f8b02646.png">
